vertexai/src/GenerateContentResponse.cs

```diff
@@ -15,20 +15,87 @@
  */
 
 using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Google.MiniJSON;
 
 namespace Firebase.VertexAI {
 
+/// <summary>
+/// The model's response to a generate content request.
+/// </summary>
 public readonly struct GenerateContentResponse {
-  public IEnumerable<Candidate> Candidates { get; }
+  private readonly ReadOnlyCollection<Candidate> _candidates;
+
+  /// <summary>
+  /// A list of candidate response content, ordered from best to worst.
+  /// </summary>
+  public IEnumerable<Candidate> Candidates =>
+      _candidates ?? new ReadOnlyCollection<Candidate>(new List<Candidate>());
+
+  /// <summary>
+  /// A value containing the safety ratings for the response, or,
+  /// if the request was blocked, a reason for blocking the request.
+  /// </summary>
   public PromptFeedback? PromptFeedback { get; }
+
+  /// <summary>
+  /// Token usage metadata for processing the generate content request.
+  /// </summary>
   public UsageMetadata? UsageMetadata { get; }
 
-  // Helper properties
-  // The response's content as text, if it exists
-  public string Text { get; }
+  /// <summary>
+  /// The response's content as text, if it exists
+  /// </summary>
+  public string Text {
+    get {
+      // Concatenate all of the text parts from the first candidate.
+      return string.Join(" ",
+          Candidates.FirstOrDefault().Content.Parts
+          .OfType<ModelContent.TextPart>().Select(tp => tp.Text));
+    }
+  }
+
+  /// <summary>
+  /// Returns function calls found in any Parts of the first candidate of the response, if any.
```

Comment:
Should `Part`s be in single quotes or linked to since it's a type?

Reply:
Honestly, not sure how doxygen will look like, but for consistency for the time being until I get around to testing it, I'll aim to single quote around the types, since that is usually correct.